src/ResourceManager/RecoveryServices.Backup/Commands.RecoveryServices.Backup.Helpers/Validations/PolicyValidations.cs

```diff
@@ -55,7 +55,15 @@ public partial class PolicyHelpers
             {
                 ValidateRetentionAndBackupTimes(schPolicy.ScheduleRunTimes, ltrPolicy.WeeklySchedule.RetentionTimes);
 
-                // validate days of week - TBD
+                // count of daysOfWeek should match for weekly schedule
+                if(ltrPolicy.WeeklySchedule.DaysOfTheWeek.Count != schPolicy.ScheduleRunDays.Count)
+                {
+                    throw new ArgumentException(
+                        "DaysOfTheWeek in WeeklySchedule of retention policy must be same as ScheduleRunDays in SchedulePolicy");
```

Comment:
Please create an entry for this in resx file and use the Resources class reference here.